### PR TITLE
fix(inputs.exec): Clean up grandchildren processes

### PR DESCRIPTION
**Pull request by powersj**
fixes: #13935

### CODE REVIEW

File: plugins/inputs/exec/run_notwinodws.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+) ([]byte, []byte, error) {
+	splitCmd, err := shellquote.Split(command)
+	if err != nil || len(splitCmd) == 0 {
+		return nil, nil, fmt.Errorf("exec: unable to parse command: %w", err)
+	}
+
+	cmd := osExec.Command(splitCmd[0], splitCmd[1:]...)
+	if runtime.GOOS != "windows" {
+		cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	}
```
Comment (srebhan) [inline plugins/inputs/exec/run_notwinodws.go:32]: Hmmm why did you split the code? Just to set the process attribute? If so, why not use the code of `run_notwindows.go` with the `runtime.GOOS` statement? I think that's the only difference between the two...

**Review by srebhan — COMMENTED** [review]
Question from my side...

File: plugins/inputs/exec/run_notwinodws.go (hunk 1 above)
Comment (powersj) [reply plugins/inputs/exec/run_notwinodws.go:32]: Unfortunately, Windows' `SysProcAttr` does not have the `Setpgid` attribute that we set here, so builds will fail. An Linux's `SysProcAttr` does not have the `CreationFlags`

[linux SysProcAttr](https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/syscall/exec_linux.go;l=62)
[winodows SysProcAttr](https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/syscall/exec_windows.go)

Comment (srebhan) [reply plugins/inputs/exec/run_notwinodws.go:32]: Ok, but then you should remove the `if runtime.GOOS != "windows" {` part for the non-Windows files... :-)

Comment (powersj) [reply plugins/inputs/exec/run_notwinodws.go]: ah yes :) done!

**Review by srebhan — APPROVED** [review]
Thanks @powersj!

**Pull request merged**